### PR TITLE
Machine healthcheck tests

### PR DESCRIPTION
**Pull request by cynepco3hahue**
Tests:
- [Feature:MachineHealthCheck] MachineHealthCheck controller
  with node-unhealth-conditions configmap should delete unhealthy machine
- [Feature:MachineHealthCheck] MachineHealthCheck controller
  should delete the unhealthy machine
    
Copy tests from the https://github.com/openshift/machine-api-operator/pull/204

### CODE REVIEW

**Comment by cynepco3hahue** [comment]
@enxebre @ingvagabund Can you please run tests?

**Comment by cynepco3hahue** [comment]
@ingvagabund Does OpenShift CI has some problems, I can see
```
2019/03/13 07:45:23 Build src failed, printing logs:
Pulling image "registry.svc.ci.openshift.org/ci/clonerefs@sha256:835268c14fbcad843bc9752ba579ec3be2e7d19c94f6dfc627a3dacfeaf1afce" ...
error: error pulling image registry.svc.ci.openshift.org/ci/clonerefs@sha256:835268c14fbcad843bc9752ba579ec3be2e7d19c94f6dfc627a3dacfeaf1afce: received unexpected HTTP status: 500 Internal Server Error
```

**Comment by ingvagabund** [comment]
> @ingvagabund Does OpenShift CI has some problems, I can see

This happens from time to time

**Comment by ingvagabund** [comment]
@cynepco3hahue the CI is currently broken. The issue is under investigation and being resolved.

**Comment by cynepco3hahue** [comment]
@ingvagabund Great thanks for info.

**Comment by cynepco3hahue** [comment]
@ingvagabund I can see that `k8s-e2e` fails on timeout 
```
E0313 11:29:28.355551   16992 utils.go:27] Error querying api for Deployment object "clusterapi-manager-controllers": deployments.apps "clusterapi-manager-controllers" not found, retrying...
E0313 11:29:28.357045   16992 utils.go:27] Error querying api for Deployment object "clusterapi-manager-controllers": deployments.apps "clusterapi-manager-controllers" not found, retrying...
E0313 11:29:28.357071   16992 utils.go:51] Error getting deployment: error getting deployment "clusterapi-manager-controllers": timed out waiting for the condition
E0313 11:29:28.357081   16992 utils.go:61] Error checking isDeploymentAvailable: timed out waiting for the condition
```
Do you have an idea of why it happens?

**Comment by ingvagabund** [comment]
>  I can see that k8s-e2e fails on timeout

flaky test

File: Gopkg.toml [hunk 1]
```diff
@@ -49,6 +49,10 @@ required = [
   name = "github.com/openshift/cluster-autoscaler-operator"
   branch = "master"
 
+[[constraint]]
+  name = "github.com/openshift/machine-api-operator"
+  revision = "9650e16c98802a4b57b7551201b0973fcae2f738"
```
Comment (ingvagabund) [inline Gopkg.toml]: This will cause a trouble in the mao repo itself. Though, we still may override the constraint. So good to go.

**Comment by ingvagabund** [comment]
@cynepco3hahue thanks for the PR!!!

File: pkg/e2e/machinehealthcheck/machinehealthcheck.go [hunk 2]
```diff
@@ -0,0 +1,230 @@
...
+
+		listOptions := runtimeclient.ListOptions{
+			Namespace: e2e.TestContext.MachineApiNamespace,
+		}
+		listOptions.SetLabelSelector(fmt.Sprintf("%s=", e2e.NodeWorkerLabel))
+		workers := &corev1.NodeList{}
+		err = client.List(context.TODO(), &listOptions, workers)
+		Expect(err).ToNot(HaveOccurred())
+
+		numberOfReadyWorkers = 0
```
Comment (enxebre) [inline pkg/e2e/machinehealthcheck/machinehealthcheck.go]: This deserves its own function and put it into utils. There's a library meant to be reusable in infra/utils. So then ginkgo e2e user stories are abstracted and easy to read with out caring about details

File: pkg/e2e/machinehealthcheck/machinehealthcheck.go [hunk 3]
```diff
@@ -0,0 +1,230 @@
...
+		Expect(workerNode).ToNot(BeNil())
+
+		listOptions = runtimeclient.ListOptions{
+			Namespace: e2e.TestContext.MachineApiNamespace,
+		}
+		machineList := &mapiv1beta1.MachineList{}
+		err = client.List(context.TODO(), &listOptions, machineList)
+		Expect(err).ToNot(HaveOccurred())
+
+		for i, m := range machineList.Items {
```
Comment (enxebre) [inline pkg/e2e/machinehealthcheck/machinehealthcheck.go]: This deserves its own function and put it into utils. There's a library meant to be reusable in infra/utils. So then ginkgo e2e user stories are abstracted and easy to read with out caring about details

File: pkg/e2e/machinehealthcheck/machinehealthcheck.go [hunk 4]
```diff
@@ -0,0 +1,230 @@
...
+			if m.Status.NodeRef != nil && m.Status.NodeRef.Name == workerNode.Name {
+				workerMachine = &machineList.Items[i]
+				glog.V(2).Infof("Worker machine %s", workerMachine.Name)
+			}
+		}
+		Expect(workerMachine).ToNot(BeNil())
+
+		if strings.Contains(workerMachine.Name, "kubemark") {
```
Comment (enxebre) [inline pkg/e2e/machinehealthcheck/machinehealthcheck.go]: this needs a TODO

**Comment by enxebre** [comment]
@cynepco3hahue thanks! I didn't have the chance to review that, so dropped some comments for the record so we can translate into follow ups improvements

**Comment by cynepco3hahue** [comment]
@enxebre Created the PR, for all your comments :smiley: 
https://github.com/openshift/cluster-api-actuator-pkg/pull/51

**Pull request merged**